packages/trigger-client/src/TriggerClient.ts

```diff
@@ -0,0 +1,283 @@
+  ) {
+    const cancelOrdersParams: OrderCancellationParams = {
+      digests: params.digests,
+      nonce: params.nonce ?? getOrderNonce(),
+      productIds: params.productIds,
+      subaccountName: params.subaccountName,
+      subaccountOwner: params.subaccountOwner,
+    };
+    const tx = getVertexEIP712Values('cancel_orders', cancelOrdersParams);
+
+    const executeParams: TriggerServerExecuteRequestByType['cancel_orders'] = {
+      signature: await this.sign(
+        'cancel_orders',
+        params.verifyingAddr,
+        await this.getChainIdIfNeeded(params),
+        cancelOrdersParams,
+      ),
+      tx,
+    };
+
+    return this.execute('cancel_orders', executeParams);
+  }
+
+  async cancelProductOrders(
+    params: TriggerExecuteRequestParamsByType['cancel_product_orders'],
+  ) {
+    const cancelProductOrdersParams: ProductOrdersCancellationParams = {
+      nonce: params.nonce ?? getOrderNonce(),
+      productIds: params.productIds,
+      subaccountName: params.subaccountName,
+      subaccountOwner: params.subaccountOwner,
+    };
+    const tx = getVertexEIP712Values(
+      'cancel_product_orders',
+      cancelProductOrdersParams,
+    );
+
+    const executeParams: TriggerServerExecuteRequestByType['cancel_product_orders'] =
+      {
+        signature: await this.sign(
+          'cancel_product_orders',
+          params.verifyingAddr,
+          await this.getChainIdIfNeeded(params),
+          cancelProductOrdersParams,
+        ),
+        tx,
+      };
+
+    return this.execute('cancel_product_orders', executeParams);
+  }
+
+  /*
+  Queries
+   */
+  async listTriggerOrders(
+    params: TriggerQueryRequestParamsByType['list_trigger_orders'],
+  ): Promise<TriggerQueryResponseByType['list_trigger_orders']> {
+    const signatureParams: ListTriggerOrdersParams = {
+      // Default to 90 seconds from now if no recvTime is provided
+      recvTime:
+        params.recvTime?.toFixed() ?? (Date.now() + 90 * 1000).toFixed(),
+      subaccountName: params.subaccountName,
+      subaccountOwner: params.subaccountOwner,
+    };
+
+    const tx = getVertexEIP712Values('list_trigger_orders', signatureParams);
+    const signature = await this.sign(
+      'list_trigger_orders',
+      params.verifyingAddr,
+      await this.getChainIdIfNeeded(params),
+      signatureParams,
+    );
+
+    const queryParams: TriggerServerQueryRequestByType['list_trigger_orders'] =
+      {
+        limit: params.limit,
+        max_update_time: params.maxUpdateTime,
+        pending: params.pending,
+        product_ids: params.productIds,
+        signature,
+        tx,
+      };
+
+    const baseResponse = await this.query('list_trigger_orders', queryParams);
+
+    const orders: TriggerOrderInfo[] =
+      baseResponse.orders.map(mapServerOrderInfo);
+
+    return {
+      orders,
+    };
+  }
+
+  /*
+  Base Fns
+   */
+  protected async sign<T extends SignableRequestType>(
```

Comment:
could we extract these in a base class?

Reply:
I think if we could share between engine and trigger service it's worth doing.

Reply:
Ah i see, i get it now. Tbh if input / output are different thinks make sense to leave it separate, my only concern was around the logic itself since its duplicated. But i prefer this over complicating the input / output types.

Reply:
Actually signing could deff be shared no?

Reply:
> Yeah it might be more readable to just duplicate

agree, plus we don't really have a great pattern for sharing a base across different clients and tbh i'd probably do the same thing we did here when it comes time to integrate this onto the python sdk